docs/source/en/migration.md

```diff
@@ -1 +1,300 @@
+- Only support Node.js 8 and above.
+- Better error stack messages without [co], approximately 30% performance improvement (do not include the performance improvement brought by Node), see [benchmark](https://eggjs.github.io/benchmark/plot/) for more details.
+
+One of the Egg's concept is `progressive`, hence we provide progressive programming experiences to developers.
+
+- [Quick upgrade](#quick_upgrade)
+- [Plugin update](#plugin)
+- [Further upgradation](#forther_upgradation)
+- [Upgrade documents for Plugin developers](#upgrade_documents_for_plugin_developers)
+
+## Quick upgrade
+- Use the latest Node LTS version (`>=8.9.0`).
+- Change `egg` version to `^2.0.0` in `package.json`.
+- Check if included plugins are the latest version (optional).
+- Reinstall the dependencies, and run unit tests again.
+
+**Done! Upgrade finished without any code changes.**
+
+
+## Plugin update
+
+### egg-multipart
+
+`yield parts` nees to change to `await parts()` or `yield poarts()`
+
+```js
+// old
+const parts = ctx.multipart();
+while ((part = yield parts) != null) {
+  // do something
+}
+
+// yield parts() also work
+while ((part = yield parts()) != null) {
+  // do something
+}
+
+// new
+const parts = ctx.multipart();
+while ((part = await parts()) != null) {
+  // do something
+}
+```
+
+- [egg-multipart#upload-multiple-files](https://github.com/eggjs/egg-multipart#upload-multiple-files)
+
+### egg-userrole
+
+DO NOT support 1.x role definition, because koa-roles is no longer compatible. 
+The `Context` has changed from `this` to the first argument `ctx`, the original `scope` now is the second the argument.
+
+```js
+// old
+app.role.use('user', function() {
+  return !!this.user;
+});
+
+// new
+app.role.use((ctx, scope) => {
+  return !!ctx.user
+});
+
+app.role.use('user', ctx => {
+  return !!ctx.user;
+});
+```
+
+- [koajs/koa-roles#13](https://github.com/koajs/koa-roles/pull/13)
+- [eggjs/egg-userrole#9](https://github.com/eggjs/egg-userrole/pull/9)
+
+## Further upgradation
+Due to the complete compatibility to Egg 1.x, we can finish upgrade quickly. 
+
+But in order to keep the code pattern consistant, as well as a better performance improvement and more developer-fridently error stack logs, thus we suggest deveopers to do a further upgrade:
+
+- Use recommended code style, see [Style guide](./style-guide.md)
+- [Use Koa style middleware](#use-koa2-style-middleware)
+- [Change `yieldable` to `awaitable` in function invoke](#yieldable-to-awaitable)
+
+### Use Koa2 style middleware
+
+> 2.x is compatible to 1.x style middleware, so it's still functional without any changes.
+
+- Use Koa 2's `(ctx, next)` arguments style in callback function
+  - The 1st argument is `ctx`, means context, it is the instance of [Context](./basics/extend.md#Context)
+  - The 2nd argument is `next`, use await to execute it for the coming logics.
+- Do not recommend to use `async (ctx, next) => {}`, this is preventing anonymous function in error stack logs 
```

Comment:
this line should be rewritten to: use `async (ctx, next) => {}` is not recommended, which prevents anonymous function in error stack.

Reply:
Changed to 
> Using `async (ctx, next) => {}` is not recommended, which prevents anonymous function in error stack.